src/modules/navigator/mission_base.cpp

```diff
@@ -1377,3 +1378,12 @@ bool MissionBase::canRunMissionFeasibility()
 	       (_geofence_status_sub.get().geofence_id == _mission.geofence_id) &&
 	       (_geofence_status_sub.get().status == geofence_status_s::GF_STATUS_READY);
 }
+
+void MissionBase::updateMissionAltAfterHomeChanged()
+{
+	if (_navigator->get_home_position()->update_count > _home_update_counter) {
+		_navigator->get_position_setpoint_triplet()->current.alt = get_absolute_altitude_for_item(_mission_item);
+		_navigator->set_position_setpoint_triplet_updated();
+		_home_update_counter = _navigator->get_home_position()->update_count;
```

Comment:
We should also check if the previous and next setpoint are valid and finite as well, if they are, correct them as well.

Reply:
> We should also check if the previous and next setpoint are valid and finite as well, if they are, correct them as well.


@KonradRudin This is not necessary because when the next item starts execution, the altitude will be recalculated based on the current home altitude. Therefore, here we only need to consider the target altitude of the task currently being executed.

Reply:
But if i'm not mistaken, it makes a first order hold of the altitude for fixed wing and tracks that slope. If you don't change the previous altitude this calculation will be wrong as well. Not sure if the next altitude is actually used somewhere. https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/fw_pos_control/FixedwingPositionControl.cpp#L1062-L1091 . For multicopter it should be fine.